api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/global/raw/ATRCSurveyGroupComplete.java

```diff
@@ -30,34 +33,88 @@ public ATRCSurveyGroupComplete() {
-        val transectsGroupedBySurveyGroup = stagedSurveyTransects.stream().collect(
-                Collectors.groupingBy(stagedSurveyMethod -> new Tuple3(stagedSurveyMethod.getSiteCode(),
+        val transectsGroupedBySurveyGroup = stagedSurveyTransects.stream()
+                .collect(Collectors.groupingBy(stagedSurveyMethod -> new Tuple3(stagedSurveyMethod.getSiteCode(),
                         stagedSurveyMethod.getDate(), stagedSurveyMethod.getDepth())));
-        return transectsGroupedBySurveyGroup.entrySet()
-                                     .stream()
-                                     .map((entry) -> validateSurveyGroup(job, entry.getKey(), entry.getValue()))
-                                     .reduce(Validated.valid("survey group is complete: nothing to validate"), (acc,
-                                      validator) ->
-                        acc.combine(Monoids.stringConcat, validator)
-                );
+        return transectsGroupedBySurveyGroup.entrySet().stream()
+                .map((entry) -> validateSurveyGroup(job, entry.getKey(), entry.getValue()))
+                .reduce(Validated.valid("survey group is complete: nothing to validate"),
+                        (acc, validator) -> acc.combine(Monoids.stringConcat, validator));
     }
 
     private Validated<StagedRowError, String> validateSurveyGroup(StagedJob job, Tuple3 surveyGroupKey,
-     List<StagedSurveyTransect> transects) {
-        val surveyNums = transects.stream()
-                              .map(StagedSurveyTransect::getSurveyNum)
-                              .collect(Collectors.toList());
-
-        if (surveyGroupComplete(surveyNums)) {
-            return Validated.valid("surveyGroup " + surveyGroupKey + ": is complete");
-        } else {
-            return invalid(job.getId(),
-                    surveyGroupKey + " has incorrect set of surveyNums: " + surveyNums,
+            List<StagedSurveyTransect> transects) {
+        
+                List<String> surveyNums = transects.stream().map(StagedSurveyTransect::getSurveyNum).collect(Collectors.toList());
+
+        if (!surveyGroupComplete(surveyNums)) {
+            return invalid(job.getId(), surveyGroupKey + " has incorrect set of surveyNums: " + surveyNums,
                     ValidationLevel.BLOCKING);
         }
+
+        val surveys = transects.stream().collect(Collectors.groupingBy(StagedSurveyTransect::getSurveyNum));
+
+        List<Validated<StagedRowError,String>> result = new ArrayList<Validated<StagedRowError,String>>();
+
+        surveys.keySet().stream().forEach(key -> {
+
+            List<StagedSurveyTransect> surveyRecords = surveys.get(key);
+            Map<String, List<StagedSurveyTransect>> surveyByMethod = surveyRecords.stream()
+                    .collect(Collectors.groupingBy(StagedSurveyTransect::getMethod));
+
+            surveyByMethod.keySet().stream().forEach(method -> {
```

Comment:
Perhaps just check methods 1 and 2 specifically - there are other combinations - and there are no rules specified for those methods:

```
select distinct sm.method_id, block_num
  from nrmn.survey_method sm
  join nrmn.survey s on (s.survey_id = sm.survey_id and s.program_id = 2)
```

method_id | block_num
-- | --
3 | 0
7 | 2
10 | 1
1 | 1
11 | NULL
4 | NULL
2 | 2
0 | NULL
6 | NULL
2 | 1
3 | NULL
5 | NULL
10 | 2
1 | 2


@bpasquer - can you confirm.